kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/BrokerAddressPatternUtils.java

```diff
@@ -110,8 +116,14 @@ static String replaceLiteralNodeId(String brokerAddressPattern, int nodeId) {
         return LITERAL_NODE_ID_PATTERN.matcher(brokerAddressPattern).replaceAll(Integer.toString(nodeId));
     }
 
-    static Pattern createNodeIdCapturingRegex(String brokerAddressPattern) {
-        var partsQuoted = Arrays.stream(brokerAddressPattern.split(Pattern.quote(LITERAL_NODE_ID))).map(Pattern::quote);
-        return Pattern.compile(partsQuoted.collect(Collectors.joining("(\\d+)")), Pattern.CASE_INSENSITIVE);
+    static String createNodeIdCapturingRegexPattern(String brokerAddressPattern, String virtualClusterName) {
+        String clusterReplaced = replaceVirtualClusterPattern(brokerAddressPattern, virtualClusterName);
+        // the -1 split limit produces an empty final item in the array if the node id replacement is at the end of the string
```

Comment:
Think I've fixed a pre-existing bug here. If `$(nodeId)` is at the end of the advertiserBrokerAddressPattern String then the `(d+)` was not added on the end as we are joining a single item.

Reply:
there are new unit tests covering this case

Reply:
split's default is surprising:

```
System.out.println(Arrays.toString("abc".split("a")));
[, bc]
System.out.println(Arrays.toString("abc".split("c")));
[ab]
```

vs.
```
System.out.println(Arrays.toString("abc".split("c", -1)));
[ab, ]
```